src/components/common/ProductBox/ProductBox.js

```diff
@@ -61,9 +67,14 @@ const ProductBox = ({
       <div className={styles.line}></div>
       <div className={styles.actions}>
         <div className={styles.outlines}>
-          <Button variant='outline'>
-            <FontAwesomeIcon icon={faHeart} className={heart ? styles.heart : ''}>
-              Favorite
+          <Button
+            className={heart ? styles.heart : ''}
+            variant='outline'
+            onClick={addToFavouriteHandler}
+          >
+            {/* <FontAwesomeIcon onClick={() => {heart ? removeFromFavourite({ id }) : addToFavourite({ id });}} id={id} icon={faHeart}> */}
+            <FontAwesomeIcon onClick={addToFavourite({ id })} id={id} icon={faHeart}   >
```

Comment:
Zamiast przypisać etodę do obsługi zdarzenia `click` to ją wywołaś, co skurkotwało zapętleniem z uwagi na ciagły rerender. Ale obsługe przenieś do `<Buttoon>` tam w metodzie addToFavouriteHandler (która obecnie nie wiele robi) wywołaj metodę `addToFavourite`